### PR TITLE
case insensitive login

### PR DESCRIPTION
**Pull request by acumarav**
Support case insensitive login/password reset
Datanode password reset
Closes OHDSI/ArachneCentralAPI#677

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseAuthenticationController.java [hunk 1]
```diff
@@ -120,6 +120,7 @@ public BaseAuthenticationController(Authenticator authenticator,
             throws AuthenticationException {
 
         JsonResult<CommonAuthenticationResponse> jsonResult;
+        authenticationRequest.normalizeLogin();
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseAuthenticationController.java:123]: why I should call this?

File: src/main/java/com/odysseusinc/arachne/portal/api/v1/dto/converters/BaseUserDtoToUserConverter.java [hunk 2]
```diff
@@ -36,7 +36,8 @@
     public U convert(CU dto) {
 
         U user = newUser();
-        user.setEmail(dto.getEmail());
+        //user.setEmail(dto.getEmail());
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/portal/api/v1/dto/converters/BaseUserDtoToUserConverter.java:39]: remove comment?

File: src/main/resources/db/migration/V20200526141500__lowercase_email_username.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+
+update users_data set
```
Comment (wivern) [inline src/main/resources/db/migration/V20200526141500__lowercase_email_username.sql:2]: Why username should be in a lower case? I guess there are authentication providers that might be case-sensitive.

**Comment by acumarav** [comment]
done

**Pull request closed without merging**